json-schema-provider/src/main/java/io/confluent/kafka/schemaregistry/json/diff/NumberSchemaDiff.java

```diff
@@ -97,10 +98,16 @@ static void compare(final Context ctx, final NumberSchema original, final Number
         ctx.addDifference("exclusiveMinimum", EXCLUSIVE_MINIMUM_DECREASED);
       }
     }
-    if (!Objects.equals(original.getMultipleOf(), update.getMultipleOf())) {
-      if (original.getMultipleOf() == null && update.getMultipleOf() != null) {
+    BigDecimal updateMultipleOf = update.getMultipleOf() != null
+        ? new BigDecimal(update.getMultipleOf().toString())
+        : null;
+    BigDecimal originalMultipleOf = original.getMultipleOf() != null
+        ? new BigDecimal(original.getMultipleOf().toString())
+        : null;
```

Comment:
We convert to BigDecimal so we can accurately tell if the new `multipleOf` value is a multiple of the old `multipleOf` value when the values are decimals.  For example, to tell if `0.002` is a multiple of `0.001`.  Otherwise this ends being a modulo of floating point numbers, which is not always accurate.